### PR TITLE
Move prepare db to migration from MigrationBuilder to MigrationService

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️/❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/Service/MigrationService.php [hunk 1]
```diff
@@ -548,16 +562,52 @@ public function generatorTemplateFiles(array $value = []): void
 
     public function up(MigrationInterface $migration): void
     {
+        $this->beforeMigrate();
         $migration->up($this->createBuilder());
+        $this->afterMigrate();
         $this->addMigrationHistory(get_class($migration));
+//        $this->afterMigrate();
```
Comment (terabytesoftw) [inline src/Service/MigrationService.php:569]:
```suggestion
        $this->afterMigrate();
```

File: src/Service/MigrationService.php [hunk 2]
```diff
@@ -548,16 +562,52 @@ public function generatorTemplateFiles(array $value = []): void
...
+    {
+        if ($this->queryCacheEnabled) {
+            $this->queryCache->setEnable(true);
+        }
+
+        if ($this->schemaCacheEnabled) {
+            $this->schemaCache->setEnable(true);
+        }
+
+        $this->queryCache->info = [];
```
Comment (terabytesoftw) [inline src/Service/MigrationService.php:607]:
```suggestion
```

File: src/Service/MigrationService.php [hunk 3]
```diff
@@ -548,16 +562,52 @@ public function generatorTemplateFiles(array $value = []): void
 
     public function up(MigrationInterface $migration): void
     {
+        $this->beforeMigrate();
         $migration->up($this->createBuilder());
+        $this->afterMigrate();
```
Comment (terabytesoftw) [inline src/Service/MigrationService.php:567]:
```suggestion
```

**Comment by vjik** [comment]
waiting https://github.com/yiisoft/db/pull/212

**Comment by terabytesoftw** [comment]
@vjik remove comment //, and adjust the PR to merge it.

**Review by terabytesoftw — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**